pkg/compiler/analysis.go

```diff
@@ -284,12 +285,31 @@ func (c *codegen) analyzeFuncUsage() funcUsage {
 				if isMain && n.Name.IsExported() || isInitFunc(n) || isDeployFunc(n) {
 					diff[name] = true
 				}
+				if isMain && n.Name.IsExported() {
+					if n.Type.Params.List != nil {
+						for i, param := range n.Type.Params.List {
+							if param.Names == nil {
+								c.prog.Err = fmt.Errorf("exported method %s is not allowed to have unnamed parameter", n.Name)
+								return false // Program is invalid.
```

Comment:
I'm not exactly excited about this approach (`analyzeFuncUsage` doing any kind of checks and producing errors), but I don't see any obviously better way.

Reply:
`analyzeFuncUsage` is the earliest place where code traversing occurs. Maybe we can move this check to a later traversing stage, but I think it's better to catch this error as fast as possible.